code/modules/assembly/flash.dm

```diff
@@ -8,45 +8,35 @@
 	throwforce = 0
 	w_class = WEIGHT_CLASS_TINY
 	materials = list(MAT_METAL = 300, MAT_GLASS = 300)
-
-	crit_fail = 0     //Is the flash burnt out?
+	crit_fail = FALSE     //Is the flash burnt out?
 	var/times_used = 0 //Number of times it's been used.
+	var/burnout_resistance = 0
 	var/last_used = 0 //last world.time it was used.
+	var/cooldown = 0
+	var/last_use = 0

```

Comment:
What's the difference from last_used? If there have to be both you might want to rename one of these to avoid confusion